sw/device/lib/dif/dif_usbdev.c

```diff
@@ -0,0 +1,1061 @@
+  usbdev->base_addr = config->base_addr;
+
+  // Initialize the free buffer pool
+  if (!buffer_pool_init(&usbdev->buffer_pool)) {
+    return kDifUsbdevError;
+  }
+
+  // No active buffer operation
+  buffer_op_init(usbdev);
+
+  // Determine the value of the PHY_CONFIG register.
+  uint32_t phy_config_val = 0;
+
+  if (config->differential_rx == kDifUsbdevToggleEnable) {
+    phy_config_val = bitfield_set_field32(
+        phy_config_val, (bitfield_field32_t){
+                            .mask = 1,
+                            .index = USBDEV_PHY_CONFIG_RX_DIFFERENTIAL_MODE,
+                            .value = 1,
+                        });
+  }
+
+  if (config->differential_tx == kDifUsbdevToggleEnable) {
+    phy_config_val = bitfield_set_field32(
+        phy_config_val, (bitfield_field32_t){
+                            .mask = 1,
+                            .index = USBDEV_PHY_CONFIG_TX_DIFFERENTIAL_MODE,
+                            .value = 1,
+                        });
+  }
+
+  if (config->single_bit_eop == kDifUsbdevToggleEnable) {
+    phy_config_val = bitfield_set_field32(
+        phy_config_val,
+        (bitfield_field32_t){
+            .mask = 1, .index = USBDEV_PHY_CONFIG_EOP_SINGLE_BIT, .value = 1,
+        });
+  }
+
+  if (config->power_sense_override == kDifUsbdevPowerSenseOverridePresent) {
+    phy_config_val = bitfield_set_field32(
+        phy_config_val, (bitfield_field32_t){
+                            .mask = 1,
+                            .index = USBDEV_PHY_CONFIG_OVERRIDE_PWR_SENSE_EN,
+                            .value = 1,
+                        });
+    phy_config_val = bitfield_set_field32(
+        phy_config_val, (bitfield_field32_t){
+                            .mask = 1,
+                            .index = USBDEV_PHY_CONFIG_OVERRIDE_PWR_SENSE_VAL,
+                            .value = 1,
+                        });
+  } else if (config->power_sense_override ==
+             kDifUsbdevPowerSenseOverrideNotPresent) {
+    phy_config_val = bitfield_set_field32(
+        phy_config_val, (bitfield_field32_t){
+                            .mask = 1,
+                            .index = USBDEV_PHY_CONFIG_OVERRIDE_PWR_SENSE_EN,
+                            .value = 1,
+                        });
+  }
+
+  // Write configuration to PHY_CONFIG register
+  mmio_region_write32(usbdev->base_addr, USBDEV_PHY_CONFIG_REG_OFFSET,
+                      phy_config_val);
+
+  return kDifUsbdevOK;
+}
+
+dif_usbdev_result_t dif_usbdev_fill_available_fifo(dif_usbdev_t *usbdev) {
+  if (usbdev == NULL) {
+    return kDifUsbdevBadArg;
+  }
+
+  // Remove entries from the pool and write them to the AV FIFO until it is full
+  while (!mmio_region_get_bit32(usbdev->base_addr, USBDEV_USBSTAT_REG_OFFSET,
+                                USBDEV_USBSTAT_AV_FULL) &&
+         !buffer_pool_is_empty(&usbdev->buffer_pool)) {
+    uint8_t buffer_entry;
+    buffer_pool_remove(&usbdev->buffer_pool, &buffer_entry);
```

Comment:
This returns an error, yet it doesn't get acknowledged... this looks like a bug, but if it's not, consider at least writing `(void)` to make the fact that it's dropped on the ground explicit. (Applies throughout.)